src/wordlist.cc

```diff
@@ -34,21 +34,8 @@ wordlistAdd(wordlist ** list, const char *key)
 void
 wordlistCat(const wordlist * w, MemBuf * mb)
 {
-    while (nullptr != w) {
-        mb->appendf("%s\n", w->key);
-        w = w->next;
-    }
-}
-
-SBufList
-ToSBufList(wordlist *wl)
-{
-    SBufList rv;
-    while (wl != nullptr) {
-        rv.push_back(SBuf(wl->key));
-        wl = wl->next;
-    }
-    return rv;
+    for (const auto &word: *w)
+        mb->appendf("%s\n", word);
```

Comment:
A few other places can be converted to use the new iterators, but I do not think they should be converted (in this PR). IMO,. it makes sense to convert this internal worldlist code from dogfooding point of view, but other code is not trivial to convert and/or should be refactored to avoid worldlist in the first place.

This comment does not request any changes.